src/Util/Datastore.php

```diff
@@ -113,6 +113,7 @@ public function renameFile(string $oldFilePath, string $newFilePath, bool $delet
         if ($deleteFlag === false) {
             $newFilePath = $this->makeFileName($newFilePath);
         }
+        $newFilePath = FileUtilities::fixFileNameLength($newFilePath);
```

Comment:
I think we should move this to makeFileName(), so that if it is a new file or rename file operation, it will always fix the file name length

Reply:
I guess that could work, would the function ever be used externally?

Reply:
Also, delete bypasses makeFileName, so it becomes TOO long, it fails.